File: ios/MobileTracker/Models/EventContext.swift
import Foundation

/// Context information automatically added to all events
public struct EventContext: Codable {
    /// Platform identifier ("ios")
    public let platform: String
    
    /// Operating system version
    public let osVersion: String
    
    /// Application version (if available)
    public let appVersion: String?
    
    public init(platform: String, osVersion: String, appVersion: String?) {
        self.platform = platform
        self.osVersion = osVersion
        self.appVersion = appVersion
    }
}

File: android/src/main/java/com/mobiletracker/models/EventContext.kt
package com.mobiletracker.models

import kotlinx.serialization.Serializable

/**
 * Context information automatically added to all events
 * 
 * @property platform Platform identifier ("android")
 * @property osVersion Operating system version
 * @property appVersion Application version (if available)
 */
@Serializable
data class EventContext(
    val platform: String,
    val osVersion: String,
    val appVersion: String? = null
)
